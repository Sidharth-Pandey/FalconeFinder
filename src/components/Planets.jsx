import React from 'react';

const Planets = ({planets, onChange, groupId})=>{
  const optionElement = planets.map((planet)=>
  <option value={planet.name} key = {planet.name}>{planet.name} </option>
)
return(
  <select className="planet" onChange = {onChange} id= {groupId}>
    <option key= "default">select</option>
    {optionElement}
  </select>
);
}

export default Planets;
