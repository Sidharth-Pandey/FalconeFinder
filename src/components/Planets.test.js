import React from 'react';
import ReactDOM from 'react-dom';
import Planets from './Planets';
import {shallow} from 'enzyme';


describe('<Planest/> test cases',()=>{
  const props = {
    planets:   [{"name":"Donlon","distance":100},{"name":"Enchai","distance":200},{"name":"Jebing","distance":300},{"name":"Sapir","distance":400},{"name":"Lerbin","distance":500},{"name":"Pingasor","distance":600}],
    onChange : function(){},
    groupId : 0
  };
  const planets = shallow(<Planets {...props}/>);
  it('should match the snapshot', () => {
    expect(planets.html()).toMatchSnapshot();
  });

  it('it should have selectBox', () => {
    expect(planets.find('.planet').length).toEqual(1);
  });

  it('it should have all the options provided by props', () => {
    expect(planets.find('option').length).toEqual(7); //including default option select
  });

  it('it should have proper planets value', () => {
    expect(planets.find('option').at(1).props().value).toEqual('Donlon');
    expect(planets.find('option').at(2).props().value).toEqual('Enchai');
  });

})
