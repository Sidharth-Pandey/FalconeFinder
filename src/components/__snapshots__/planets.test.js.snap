// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Planest/> test cases should match the snapshot 1`] = `"<select class=\\"planet\\" id=\\"0\\"><option>select</option><option value=\\"Donlon\\">Donlon </option><option value=\\"Enchai\\">Enchai </option><option value=\\"Jebing\\">Jebing </option><option value=\\"Sapir\\">Sapir </option><option value=\\"Lerbin\\">Lerbin </option><option value=\\"Pingasor\\">Pingasor </option></select>"`;
